dev/com.ibm.ws.config/src/com/ibm/ws/config/xml/internal/ConfigEvaluator.java

```diff
@@ -1328,36 +1330,51 @@ private String processString(String value, ExtendedAttributeDefinition attrDef,
             }
         }
 
+        boolean optionFound = false;
         // Convert option value to correct case.
         String[] optionValues = attrDef.getOptionValues();
         if (optionValues != null) {
-            String[] optionLabels = attrDef.getOptionLabels();
             for (int i = 0; i < optionValues.length; i++) {
-                if (strValue.equalsIgnoreCase(optionValues[i]) || strValue.equalsIgnoreCase(optionLabels[i])) {
+                if (strValue.equalsIgnoreCase(optionValues[i])) {
                     strValue = optionValues[i];
+                    optionFound = true;
                     break;
                 }
             }
+            // only get the labels if it isn't found in the values.  Usually it will be a value.
+            if (!optionFound) {
+                String[] optionLabels = attrDef.getOptionLabels();
+                for (int i = 0; i < optionValues.length; i++) {
+                    if (strValue.equalsIgnoreCase(optionLabels[i])) {
+                        strValue = optionValues[i];
+                        optionFound = true;
+                        break;
+                    }
+                }
+            }
         }
 
-        // Special case for Boolean, since we want to validate "true" or "false" and anything is fail to validate (rather than being treated as false)
-        if (attrDef.getType() == AttributeDefinition.BOOLEAN) {
-            if (!!!("true".equalsIgnoreCase(strValue))
-                && !!!("false".equalsIgnoreCase(strValue))) {
-                Object[] inserts = new Object[] { strValue, attrDef.getID(), "false" }; // "false" is the default default
-                if (attrDef.getDefaultValue() != null && attrDef.getDefaultValue().length > 0)
-                    inserts[2] = attrDef.getDefaultValue()[0];
-                throw new AttributeValidationException(attrDef, strValue, Tr.formatMessage(tc, "error.invalid.boolean.attribute", inserts));
-            }
-        } else if (attrDef.getType() != MetaTypeFactory.PID_TYPE) {
-            // The validate method treats whitespace, commas, and
-            // backslashes specially, but we've already done all that
-            // processing and just want to validate a single value, so
-            // escape all the special characters.  (It might be less effort
-            // to just reimplement the validate method entirely.)
-            String validateResult = attrDef.validate(MetaTypeHelper.escapeValue(strValue));
-            if (validateResult != null && validateResult.length() > 0) {
-                throw new AttributeValidationException(attrDef, strValue, validateResult);
+        // If it is an option and it was found then skip the validate since it isn't needed.
+        if (!optionFound) {
```

Comment:
Small change in behavior here -- it would allow someone to use values other than 'true' or 'false' for a boolean. It won't break any existing customers, so it's fine, but it is something we'll have to support from now on.

Reply:
You could do something like:
`
<AD type="boolean"...>
  <option value="falsch"/>
 <option value="wahr"/>
</AD>
`

Previously we would have only accepted case variations of "false" and "true", but now you could do something like the above and the options would be accepted as valid values for a boolean. I don't have a big problem with it, I just wanted to note that we are adding support here that we will have to maintain forever.